source_control/git.py

```diff
@@ -368,6 +368,24 @@ def reset(git_path, module, dest):
     cmd = "%s reset --hard HEAD" % (git_path,)
     return module.run_command(cmd, check_rc=True, cwd=dest)
 
+def get_diff(module, git_path, dest, result):
+    ''' Return the difference between 2 versions '''
+    if result['before'] == None:
+        result['diff'] = { 'prepared': '>> Newly checked out %(after)s' % result }
+        return result
+    elif result['before'] != result['after']:
+        cmd = '%s diff %s %s' % (git_path, result['before'], result['after'])
+        (rc, out, err) = module.run_command(cmd, check_rc=True, cwd=dest)
+        if rc == 0 and out:
+            result['diff'] = { 'prepared': out }
+        elif rc == 0:
+            result['diff'] = { 'prepared': '>> No visual differences between %(before)s and %(after)s' % result }
+        elif err:
+            result['diff'] = { 'prepared': '>> Failed to get proper diff between %s and %s:\n>> %s' % (result['before'], result['after'], err) }
+        else:
+            result['diff'] = { 'prepared': '>> Failed to get proper diff between %(before)s and %(after)s' % result }
+    return result
```

Comment:
Programming style... Instead of operating directly on result, pass in before and after and return just the diff.  Then in calling code, do something like:

``` python
result['diff'] = get_diff(module, git_path, dest, result['before'], result['after'])
```